### ISSUE TITLE
Bug in `Promise.prototype._resolveCallback`?

### ISSUE THREAD

**Issue by overlookmotel** [petkaantonov/bluebird#1079]
I'm looking at the source code for bluebird v3.3.5. Think I might have noticed a typo/bug.

In `Promise.prototype._resolveCallback`, there is the following code:

```js
var bitField = promise._bitField;
if (BIT_FIELD_CHECK(IS_PENDING_AND_WAITING_NEG)) {
    ...
} else if (BIT_FIELD_CHECK(IS_FULFILLED)) {
    ...
} else if (BIT_FIELD_CHECK(IS_REJECTED)) {
    ...
}
```

https://github.com/petkaantonov/bluebird/blob/master/src/promise.js#L464

The `bitfield` variable is never referred to after it's defined, and the following lines call `BIT_FIELD_CHECK()` with no `bitField` argument.

I'm guessing the code above was intended to read:

```js
var bitField = promise._bitField;
if (BIT_FIELD_CHECK(IS_PENDING_AND_WAITING_NEG, bitField)) {
    ...
} else if (BIT_FIELD_CHECK(IS_FULFILLED, bitField)) {
    ...
} else if (BIT_FIELD_CHECK(IS_REJECTED, bitField)) {
    ...
}
```

... but I have no idea of the implications of this change!

**Comment by petkaantonov** [comment]
Those are not functions but macros which the build step expands

**Comment by overlookmotel** [comment]
@petkaantonov Sorry my mistake. Bluebird is a tour-de-force but the code can be pretty hard to follow for the uninitiated with all the optimizations it contains.
